src/neo-vm/ExecutionEngine.cs

```diff
@@ -1060,71 +1108,33 @@ private bool ExecuteInstruction()
-                        if (!TryPop(out StackItem x)) return false;
-                        switch (x)
-                        {
-                            case VMArray array:
-                                {
-                                    int index = key.ToInt32();
-                                    if (index < 0) return false;
-                                    Push(index < array.Count);
-                                    break;
-                                }
-                            case Map map:
-                                {
-                                    Push(map.ContainsKey(key));
-                                    break;
-                                }
-                            case Buffer buffer:
-                                {
-                                    int index = key.ToInt32();
-                                    if (index < 0) return false;
-                                    Push(index < buffer.Size);
-                                    break;
-                                }
-                            default:
-                                return false;
-                        }
+                        if (!TryPop(out CompoundType x)) return false;
+                        x.Clear();
                         break;
                     }
-                case OpCode.KEYS:
+
+                //Types
+                case OpCode.ISNULL:
                     {
-                        if (!TryPop(out Map map)) return false;
-                        Push(new VMArray(ReferenceCounter, map.Keys));
+                        if (!TryPop(out StackItem x)) return false;
+                        Push(x.IsNull);
                         break;
                     }
-                case OpCode.VALUES:
+                case OpCode.ISTYPE:
                     {
-                        IEnumerable<StackItem> values;
                         if (!TryPop(out StackItem x)) return false;
-                        switch (x)
-                        {
-                            case VMArray array:
-                                values = array;
-                                break;
-                            case Map map:
-                                values = map.Values;
-                                break;
-                            default:
-                                return false;
-                        }
-                        VMArray newArray = new VMArray(ReferenceCounter);
-                        foreach (StackItem item in values)
-                            if (item is Struct s)
-                                newArray.Add(s.Clone());
-                            else
-                                newArray.Add(item);
-                        Push(newArray);
+                        StackItemType type = (StackItemType)instruction.TokenU8;
+                        if (type == StackItemType.Any || !Enum.IsDefined(typeof(StackItemType), type))
```

Comment:
`type == StackItemType.Any` should be `ISNULL`?

Reply:
`Any` is not a real type.